From 8318852ef4f768bed31072aa7b57e11adc1f639c Mon Sep 17 00:00:00 2001
From: Andrea Adami <andrea.adami@gmail.com>
Date: Sun, 29 Jun 2014 00:44:03 +0200
Subject: [PATCH 5/6] common.h: more workarounds for klibc compatibility

Patch is addressing two issues:
* First,  Klibc doesn't have rpmatch().
* Second, Klibc lacks getline()

Fixes:
|   LD      ubi-utils/ubiformat
| .../git/ubi-utils/ubiformat.o: In function `prompt':
| .../git/./include/common.h:157: undefined reference to `getline'
| .../git/./include/common.h:164: undefined reference to `rpmatch'
| .../git/./include/common.h:157: undefined reference to `getline'
| .../git/./include/common.h:164: undefined reference to `rpmatch'

Upstream-Status: Pending

Signed-off-by: Andrea Adami <andrea.adami@gmail.com>
---
 include/common.h | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

Index: git/include/common.h
===================================================================
--- git.orig/include/common.h
+++ git/include/common.h
@@ -161,15 +161,17 @@ static inline int __rpmatch(const char *
  */
 static inline bool prompt(const char *msg, bool def)
 {
-	char *line = NULL;
-	size_t len;
+	char *line;
 	bool ret = def;
 
+	const int sizeof_line = 2;
+	line = malloc(sizeof_line);
+
 	do {
 		normsg_cont("%s (%c/%c) ", msg, def ? 'Y' : 'y', def ? 'n' : 'N');
 		fflush(stdout);
 
-		while (getline(&line, &len, stdin) == -1) {
+		while (fgets(line, sizeof_line, stdin) == NULL) {
 			printf("failed to read prompt; assuming '%s'\n",
 				def ? "yes" : "no");
 			break;
